core/datainput.gms

```diff
@@ -321,9 +321,49 @@ p_inco0(ttot,regi,teRegTechCosts)  = (1 + p_tkpremused(regi,teRegTechCosts) ) *
+$ifthen.floorscen NOT %cm_floorCostScen% == "default"
+p_oldFloorCostdata(regi,teLearn(te)) = pm_data(regi,"inco0",te) - pm_data(regi,"incolearn",te);
+$endif.floorscen
+
+*** calculate floor costs for learning technologies if historical price structure prevails
+$ifthen.floorscen %cm_floorCostScen% == "pricestruc"
+** compute maximum tech cost in 2015 for a given tech among regions
+p_maxRegTechCost2015(teRegTechCosts) = SMax(regi, p_inco0("2015",regi,teRegTechCosts));
+*take the ratio of the tech cost in 2015 and the maximum cost, and multiply with the global floor to get new floorcost that preserves the price structure
+pm_data(regi,"floorcost",teLearn(te))$(p_maxRegTechCost2015(te) ne 0) = p_oldFloorCostdata(regi,te) * p_inco0("2015",regi,te) / p_maxRegTechCost2015(te);
+* for newer data than 2015, use these
+p_maxRegTechCost2020(teRegTechCosts) = SMax(regi, p_inco0("2020",regi,teRegTechCosts));
+pm_data(regi,"floorcost",teLearn(te))$(p_maxRegTechCost2020(te) ne 0) = p_oldFloorCostdata(regi,te) * p_inco0("2020",regi,te) / p_maxRegTechCost2020(te);
+* report the new floor cost data
+p_newFloorCostdata(regi,teLearn(te))$(p_maxRegTechCost2015(te) ne 0) = p_oldFloorCostdata(regi,te) * p_inco0("2015",regi,te) / p_maxRegTechCost2015(te);
+p_newFloorCostdata(regi,teLearn(te))$(p_maxRegTechCost2020(te) ne 0) = p_oldFloorCostdata(regi,te) * p_inco0("2020",regi,te) / p_maxRegTechCost2020(te);
+$endif.floorscen
+
+*** calculate floor costs for learning technologies if there is technology transfer
+$ifthen.floorscen %cm_floorCostScen% == "techtrans"
+** compute maximum income GDP PPP per capita among regions in 2050
+p_gdppcap2050_PPP(regi) = pm_gdp("2050",regi) / pm_shPPPMER(regi) / pm_pop("2050",regi);
+p_maxPPP2050 = SMax(regi, p_gdppcap2050_PPP(regi));
+*take the ratio of the PPP income and the maximum income, and multiply with the global floor to get new floorcost that simulates tech transfer where costs are solely dependent on local wages, not on IP rent
+pm_data(regi,"floorcost",teLearn(te))$(p_maxPPP2050 ne 0) = p_oldFloorCostdata(regi,te) * p_gdppcap2050_PPP(regi) / p_maxPPP2050;
+p_newFloorCostdata(regi,teLearn(te))$(p_maxPPP2050 ne 0) = p_oldFloorCostdata(regi,te) * p_gdppcap2050_PPP(regi) / p_maxPPP2050;
+$endif.floorscen
+
+
+*** In case regionally differentiated investment costs should be used, the corresponding entries are revised:
+$if %cm_techcosts% == "REG"   pm_data(regi,"inco0",teRegTechCosts) = p_inco0("2015",regi,teRegTechCosts);
+loop(teRegTechCosts$(sameas(teRegTechCosts,"spv") ),
+$if %cm_techcosts% == "REG"   pm_data(regi,"inco0",teRegTechCosts) = p_inco0("2020",regi,teRegTechCosts);
+);
+p_maxSpvCost = SMax(regi, p_inco0("2020",regi,"spv"));
+$if %cm_techcosts% == "REG"   pm_data(regi,"inco0","storspv") = p_inco0("2020",regi,"spv")/p_maxSpvCost * pm_data(regi,"inco0","storspv");
+
+$if %cm_techcosts% == "REG"   pm_data(regi,"incolearn",teLearn(te)) = pm_data(regi,"inco0",te) - pm_data(regi,"floorcost",te) ;
 
```

Comment:
I think there has been a merge conflict here: same concept above and below line 367, starting with comment _*** In case regionally differentiated investment costs should be used, the corresponding entries are revised:_.
But the chunk below is newer, I had modified it for the non-learning-tech convergence switch.

Could you please integrate the relevant modifications in it and remove the chunk above?

Reply:
done